pkg/agent/proxy/proxier_test.go

```diff
@@ -183,12 +208,16 @@ func testClusterIP(t *testing.T, svcIP net.IP, epIP net.IP, isIPv6 bool, extraSv
 	fp.syncProxyRules()
 }
 
-func testLoadBalancer(t *testing.T, nodePortAddresses []net.IP, svcIP, ep1IP, ep2IP, loadBalancerIP net.IP, isIPv6, nodeLocalExternal bool) {
+func testLoadBalancer(t *testing.T, nodePortAddresses []net.IP, svcIP, ep1IP, ep2IP, loadBalancerIP net.IP, isIPv6, nodeLocalExternal, proxyExternalIPs bool) {
 	ctrl := gomock.NewController(t)
 	defer ctrl.Finish()
 	mockOFClient := ofmock.NewMockClient(ctrl)
 	mockRouteClient := routemock.NewMockInterface(ctrl)
-	fp := NewFakeProxier(mockRouteClient, mockOFClient, nodePortAddresses, isIPv6, true)
+	options := []proxyOptionsFn{withProxyAll}
+	if !proxyExternalIPs {
+		options = append(options, withoutProxyExternalIPs)
+	}
+	fp := NewFakeProxier(mockRouteClient, mockOFClient, nodePortAddresses, isIPv6, options...)
```

Comment:
Out of curiosity, why doesn't it pass two booleans proxyAll, proxyExternalIPs directly?

Reply:
I don't like long parameter list and it means that all call sites have to be updated every time we add a new parameter, even though most tests just want to use the default value. Plus with booleans, when you look at the call site it can be hard to figure out which argument does what. With mutator functions (e.g. `WithProxyAll`), it's quite readable. At least in the future, if we need to add a new configuration option (e.g. for ExternalIPs as you mentioned above), we won't need to grow the parameter list and update all call sites.